cpp/include/cudf/io/experimental/deletion_vectors.hpp

```diff
@@ -21,14 +21,129 @@
+   * decompression space. The `pass_read_limit` is a hint, not an absolute limit - if a single row
+   * group cannot fit within the limit given, it will still be loaded. Also note that the
+   * `pass_read_limit` does not include the memory to deserialize and construct the roaring64 bitmap
+   * deletion vector that stays alive throughout the the lifetime of the reader.
+   *
+   * @param chunk_read_limit Byte limit on the returned table chunk size, `0` if there is no limit
+   * @param pass_read_limit Byte limit on the amount of memory used for decompressing and decoding
+   * data, `0` if there is no limit
+   * @param options Parquet reader options
+   * @param serialized_roaring64 Host span of `portable` serialized 64-bit roaring bitmap
+   * @param row_group_offsets Host span of row offsets of each row group
+   * @param row_group_num_rows Host span of number of rows in each row group
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   * @param mr Device memory resource to use for device memory allocation
+   */
+  chunked_parquet_reader(
+    std::size_t chunk_read_limit,
+    std::size_t pass_read_limit,
+    parquet_reader_options const& options,
+    cudf::host_span<cuda::std::byte const> serialized_roaring64,
+    cudf::host_span<size_t const> row_group_offsets,
+    cudf::host_span<size_type const> row_group_num_rows,
+    rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+    rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref());
+
+  /**
+   * @brief Destructor, destroying the internal reader instance and the roaring bitmap deletion
+   * vector
+   */
+  ~chunked_parquet_reader();
+
+  /**
+   * @brief Check if there is any data in the given source that has not yet been read
+   *
+   * @return Boolean value indicating if there is any data left to be read
+   */
+  [[nodiscard]] bool has_next() const;
+
+  /**
+   * @brief Read a chunk of table from the Parquet source, prepend an index column to it, and
+   * filters the resultant table chunk using the 64-bit roaring bitmap deletion vector, if provided
+   *
+   * The sequence of returned tables, if concatenated by their order, guarantees to form a complete
+   * dataset as reading the entire given source at once.
+   *
+   * An empty table will be returned if the given source is empty, or all the data in the source has
+   * been read and returned by the previous calls.
+   *
+   * @return An output `cudf::table` along with its metadata
+   */
+  [[nodiscard]] table_with_metadata read_chunk();
+
+ private:
+  std::unique_ptr<cudf::io::chunked_parquet_reader> _reader;
+  std::queue<size_t> _row_group_row_offsets;
+  std::queue<size_type> _row_group_row_counts;
+  std::unique_ptr<roaring_bitmap_impl> _deletion_vector;
+  size_t _start_row;
+  bool _is_unspecified_row_group_data;
+  rmm::cuda_stream_view _stream;
+  rmm::device_async_resource_ref _mr;
+  rmm::device_async_resource_ref _table_mr;
```

Comment:
is two separate mr really needed?

Reply:
Internal thing, just makes our life easier 😄. See https://github.com/rapidsai/cudf/pull/20201#discussion_r2446139245